src/TestFramework/MSTest.Core/Attributes/ClassCleanupLifecycle.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting
+{
+    /// <summary>
+    /// When to run ClassCleanup during test execution
+    /// </summary>
+    public enum ClassCleanupLifecycle
```

Comment:
Is this naming based on some existing settings we have? I think we should choose between Sequencing and Lifecycle if there is no existing example of this and in that case Lifecycle seems easier to grasp.

Or possibly ClassCleanupExecution?

Reply:
![image](https://user-images.githubusercontent.com/5735905/132853540-d2f4d793-d2bf-4ff5-9cf6-49317cc5fea3.png)

Tried it out, this is how it looks like on the class. Which is imho bit ridiculous. Like I would have no idea what this means. :) 

These are the others that we have. 
TestDataSourceDiscoveryAttribute
ParallelizeAttribute
DoNotParallelizeAttribute
TestExtensionTypesAttribute
DiscoverInternalsAttribute

This also permits positioning it on method. Other assembly level attributes do not allow that, except for DoNotParallelize. And we have ClassCleanup, attribute, that already takes option for InheritanceBehavior , but not this one.

So imho we should: 
- [ ] only allow this new attribute on assembly level
- [ ] review the syntax
- [ ] add option to ClassCleanup attribute

Not sure about this, but it seems to more or less fit both use cases:
```cs
[assembly: ClassCleanupExecution(ExecuteAt = ClassCleanupBehavior.EndOfAssembly)]
[ClassCleanup(InheritanceBehavior.BeforeEachDerivedClass, ClassCleanupBehavior.EndOfAssembly)]
[ClassCleanup(InheritanceBehavior.BeforeEachDerivedClass, ClassCleanupBehavior.EndOfClass)][ClassCleanup(ClassCleanupBehavior.EndOfAssembly)]
```

Not sure why, but our attributes don't expose properties, so hopefully we won't have more options, because that would lead to constructor boom.